convert2rhel/pkgmanager/handlers/yum/callback.py

```diff
@@ -0,0 +1,191 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+# Copyright 2005 Duke University
+# Parts Copyright 2007 Red Hat, Inc
+#
+#   License above taken from the original code at:
+#
+#       https://github.com/rpm-software-management/yum/blob/master/yum/rpmtrans.py
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+#   License above taken from the original code at:
+#
+#       https://github.com/rpm-software-management/yum/blob/master/yum/callbacks.py
+#
+import logging
+
+from convert2rhel import pkgmanager
+
+
+loggerinst = logging.getLogger(__name__)
+"""Instance of the logger used in this module."""
+
+# We need to double inherit here, both from the callback class and the base
+# object class, just to initialize properly with `super`
+class PackageDownloadCallback(pkgmanager.DownloadProgress, object):
+    """Package download callback for YUM transaction."""
+
+    def __init__(self):
+        """Constructor for the package download progress indicator.
+
+        We initialize a few properties here for keeping track of progression of
+        the downloaded files.
+        """
+        super(PackageDownloadCallback, self).__init__()
+        # Same strategy as used in yum.rpmtrans.SimpleCliCallBack. We
+        # hold the last package name to not print it twice, avoiding
+        # spamming msgs.
+        self.last_package_seen = None
+
+    def updateProgress(self, name, frac, fread, ftime):
+        """Update and output the message that we sent to the user.
+
+        This method, in the `DownloadBaseCallback` class, is meant to be used
+        with a progress-bar to show a nice progression for the user, but we are
+        using it only to output a simple message to keep track of the packages
+        were downloaded, no need to do any fancy progress-bar.
+
+        :param name: File that is being downloaded in the transaction.
+        :type name: str
+        :param frac: A number between `0` and `1` that represents the file that is being downloaded or not.
+        :type frac: int
```

Comment:
```suggestion
        :param frac: A number between `0` and `1` that represents the percentage of the file that has been downloaded.
        :type frac: float
```
From the first part of the parameter description, I think this is what should be here but feel free to correct me if I'm wrong.